tests/k8s/test_k8s.py

```diff
@@ -941,6 +943,69 @@ async def test_flow_with_custom_gateway(logger, docker_images, tmpdir):
     core_client.delete_namespace(namespace)
 
 
+@pytest.mark.asyncio
+@pytest.mark.timeout(3600)
+@pytest.mark.parametrize(
+    'docker_images',
+    [['multiprotocol-gateway']],
+    indirect=True,
+)
+async def test_flow_multiple_protocols_gateway(logger, docker_images, tmpdir):
+    http_port = random_port()
+    grpc_port = random_port()
+    flow = Flow().config_gateway(
+        uses=f'docker://{docker_images[0]}',
+        port=[http_port, grpc_port],
+        protocol=['http', 'grpc'],
+    )
+
+    dump_path = os.path.join(str(tmpdir), 'k8s-flow-multiprotocol-gateway')
+    namespace = 'flow-multiprotocol-gateway'
+    flow.to_kubernetes_yaml(dump_path, k8s_namespace=namespace)
+
+    from kubernetes import client
+
+    api_client = client.ApiClient()
+    core_client = client.CoreV1Api(api_client=api_client)
+    app_client = client.AppsV1Api(api_client=api_client)
+    await create_all_flow_deployments_and_wait_ready(
+        dump_path,
+        namespace=namespace,
+        api_client=api_client,
+        app_client=app_client,
+        core_client=core_client,
+        deployment_replicas_expected={
+            'gateway': 1,
+        },
+        logger=logger,
+    )
+
+    gateway_pod_name = (
+        core_client.list_namespaced_pod(
+            namespace=namespace, label_selector='app=gateway'
+        )
+        .items[0]
+        .metadata.name
+    )
+    config_path = os.environ['KUBECONFIG']
+    import portforward
+
+    with portforward.forward(
```

Comment:
Testing the request using the service is also useful. I don't know if the python kubectl client can do a proxy but on local we can do something like
```
# open a proxy default port at 8081
kubectl proxy &
```
```
curl http://localhost:8081/api/v1/namespaces/${namepsace}/services/gateway:${port}/proxy
```

Reply:
I believe proxy is overkill for that. You can port forward on the service instead of the Deployment.  

`kubectl port-forward svc/NAME port1:port2`

Reply:
Anything is fine as long as the service interface is used for the test.

Reply:
done